emcc.py

```diff
@@ -2392,6 +2393,8 @@ def phase_linker_setup(options, state, newargs):
         '_wasmfs_readdir_finish',
         '_wasmfs_get_cwd',
       ]
+    if settings.ASYNCIFY == 2:
+      settings.ASYNCIFY_EXPORTS += ['__funcs_on_exit']
```

Comment:
Please add a comment here as without the discussion in the PR this looks a little surprising.